src/sudoers/tokens.rs

```diff
@@ -83,13 +83,19 @@ pub enum Meta<T> {
 }
 
 impl<T: Token> Token for Meta<T> {
-    fn construct(s: String) -> Result<Self, String> {
-        Ok(if s == "ALL" {
+    fn construct(raw: String) -> Result<Self, String> {
+        // `T` may accept whitespace resulting in `raw` having trailing whitespace which would make
```

Comment:
Unless something has changed in that part of the code in a different PR, the "tokenizer" takes care that whitespace doesn't occur here, so this comment isn't accurate (and trim_end should not be necessary)